node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Reports.java

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.provision.node;
+
+import com.yahoo.slime.Cursor;
+import com.yahoo.slime.Inspector;
+import com.yahoo.slime.ObjectTraverser;
+import com.yahoo.slime.Type;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+
+/**
+ * @author hakonhall
+ */
+// @Immutable
+public class Reports {
+    private final Map<String, Report> reports;
+
+    public Reports() { this(new TreeMap<>()); }
+    private Reports(Map<String, Report> reports) { this.reports = Collections.unmodifiableMap(reports); }
+
+    public boolean isEmpty() { return reports.isEmpty(); }
+    public Optional<Report> getReport(String id) { return Optional.ofNullable(reports.get(id)); }
+    public List<Report> getReports() { return List.copyOf(reports.values()); }
```

Comment:
Nitpick: Consider `unmodifidableList`. Since this is immutable, copying is unnecessary.

Reply:
Unfortunately unmodifiableList takes a List, while values() returns a Collection.